arsdk-rs/src/lib.rs

```diff
@@ -71,19 +114,21 @@ struct HandshakeResponse {
     status: i8,
 }
 
-fn perform_handshake(target: impl ToSocketAddrs, d2c_port: u16) -> AnyResult<HandshakeResponse> {
+fn perform_handshake(target: SocketAddr, d2c_port: u16) -> AnyResult<HandshakeResponse> {
     let handshake_message = HandshakeMessage {
         controller_name: "Drone-rs".to_string(),
         controller_type: "computer".to_string(),
         d2c_port,
+        // arstream2_client_stream_port: Some(44445),
+        // arstream2_client_control_port: Some(44446),
     };
 
     println!(
         "connecting controller {}",
         handshake_message.controller_name,
     );
 
-    let mut handshake_stream = TcpStream::connect(target)?;
+    let mut handshake_stream = retry(10, target).ok_or_else(|| anyhow!("Couldn't connect for handshake {}", target))?;
```

Comment:
We might want to switch to something better (i.e. return a Vec<Errors>)
PS: I am not a big fan of the `anyhow`